src/main/java/org/maxgamer/quickshop/Shop/ShopManager.java

```diff
@@ -441,16 +447,27 @@ private void actionTrade(@NotNull Player p, @NotNull HashMap<UUID, Info> actions
                         int invHaveSpaces = Util.countSpace(p.getInventory(), shop.getItem());
                         amount = Math.min(shopHaveItems,invHaveSpaces);
                     }else{
-                        amount = Util.countItems(p.getInventory(), shop.getItem());
+                        // should check not having items but having empty slots, cause player is trying to buy items from the shop.
+                        amount = Util.countSpace(p.getInventory(), shop.getItem());
+                    }
+                    // typed 'all', check if player has enough money than price * amount
+                    double price = shop.getPrice();
+                    double balance = eco.getBalance(p.getUniqueId());
+                    amount = Math.min(amount, (int) Math.floor(balance / price));
+                    if (amount < 1) {
+                        // when typed 'all' but player can't buy any items
+                        // @TODO: from getConfig().getString("shop.not-enough-money-for-trade-all-items");
+                        p.sendMessage("you don't have enough money!");
```

Comment:
All messages must use language file, please use MsgUtils.getMessage()

Reply:
changed to use MsgUtil.getMessage()